x/virtualgroup/types/params.go

```diff
@@ -137,3 +157,27 @@ func validateMaxStoreSizePerFamily(i interface{}) error {
 
 	return nil
 }
+
+func validateSwapInValidityPeriod(i interface{}) error {
+	v, ok := i.(uint64)
+	if !ok {
+		return fmt.Errorf("invalid parameter type: %T", i)
+	}
+	if v == 0 {
+		return fmt.Errorf("swapIn info validity period must be positive: %d", v)
+	}
+	return nil
+}
+
+func validateSPConcurrentExitNum(i interface{}) error {
```

Comment:
duplicated `validateSwapInValidityPeriod` calls. Fix - call `validateSPConcurrentExitNum` for validation.

Reply:
fixed